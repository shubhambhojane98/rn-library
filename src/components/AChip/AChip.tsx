import React, {useState} from 'react';
import {
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../utils/Common';
import ATypography from '../ATypography/ATypography';
import {TypographyVariant} from '../ATypography/ATypographyEnum';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';

interface Props {
  onPress: () => void;
  viewStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  label: string;
  selectedBackgroundColor?: string;
  selectedTextColor?: string;
  textColor?: string;
  isDisabled?: boolean;
  theme: Theme;
}

const AChip: React.FC<Props> = ({
  label,
  onPress,
  selectedBackgroundColor,
  selectedTextColor,
  textColor,
  isDisabled = false,
  textStyle = {
    paddingHorizontal: moderateScale(16, defaultScale),
    paddingVertical: moderateScale(7, defaultScale),
    fontSize: moderateScale(14, defaultScale),
    lineHeight: moderateScale(17, defaultScale),
  },
}) => {
  const [isSelected, setChipSelected] = useState(false);

  const handleChip = () => {
    if (isDisabled) {
      return;
    }
    setChipSelected(!isSelected);
    onPress();
  };
  const {colors} = useTheme();
  const bgColor = selectedBackgroundColor
    ? selectedBackgroundColor
    : colors.primary;

  return (
    <TouchableOpacity
      onPress={handleChip}
      activeOpacity={isDisabled ? 0.5 : 1}
      style={{
        ...styles.viewStyle,
        opacity: isDisabled ? 0.5 : 1,
        backgroundColor: isSelected ? bgColor : colors.white,
        borderColor: isSelected ? bgColor : colors.lightgrey,
      }}>
      <ATypography
        children={label}
        variant={TypographyVariant.SECONDARY_BOLD}
        color={
          isSelected
            ? selectedTextColor
              ? selectedTextColor
              : colors.white
            : textColor
            ? textColor
            : colors.black
        }
        fontSize={moderateScale(14, defaultScale)}
        style={textStyle}
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  viewStyle: {
    borderWidth: moderateScale(1, defaultScale),
    borderRadius: moderateScale(49, defaultScale),
    alignSelf: 'center',
    marginHorizontal: moderateScale(5, defaultScale),
    paddingHorizontal: moderateScale(16, defaultScale),
    paddingVertical: moderateScale(4, defaultScale),
  },
});
export default withTheme(AChip);
