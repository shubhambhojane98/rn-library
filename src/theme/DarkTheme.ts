import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
