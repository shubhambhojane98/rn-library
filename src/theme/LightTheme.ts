import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
