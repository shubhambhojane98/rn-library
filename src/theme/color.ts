import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
