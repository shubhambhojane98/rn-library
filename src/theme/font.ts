import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }