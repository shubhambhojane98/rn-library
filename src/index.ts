import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
