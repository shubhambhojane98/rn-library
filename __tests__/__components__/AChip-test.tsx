import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AChip from '../../src/components/AChip/AChip';
import {Color} from '../../src/theme';

it('renders correctly with default prop', () => {
  const tree = renderer
    .create(<AChip label="Whats New" onPress={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with selectedBackgroundColor & textColor & selectedTextColor prop', () => {
  const tree = renderer
    .create(
      <AChip
        label="Whats New"
        onPress={jest.fn}
        selectedBackgroundColor={Color.violet}
        textColor={Color.black}
        selectedTextColor={Color.healthcare}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
