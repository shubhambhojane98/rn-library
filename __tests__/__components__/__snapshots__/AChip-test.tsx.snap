// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderColor": "rgba(53, 53, 53, 0.2)",
      "borderRadius": 60.2,
      "borderWidth": 1.2285714285714286,
      "marginHorizontal": 6.142857142857142,
      "opacity": 1,
      "paddingHorizontal": 19.65714285714286,
      "paddingVertical": 4.914285714285715,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "AvenirNextLTPro-Bold",
          "fontSize": 21.13142857142857,
          "fontWeight": "bold",
        },
        Object {
          "color": "#231F20",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "fontSize": 17.2,
          "lineHeight": 20.885714285714286,
          "paddingHorizontal": 19.65714285714286,
          "paddingVertical": 8.6,
        },
      ]
    }
  >
    Whats New
  </Text>
</View>
`;

exports[`renders correctly with selectedBackgroundColor & textColor & selectedTextColor prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderColor": "rgba(53, 53, 53, 0.2)",
      "borderRadius": 60.2,
      "borderWidth": 1.2285714285714286,
      "marginHorizontal": 6.142857142857142,
      "opacity": 1,
      "paddingHorizontal": 19.65714285714286,
      "paddingVertical": 4.914285714285715,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "AvenirNextLTPro-Bold",
          "fontSize": 21.13142857142857,
          "fontWeight": "bold",
        },
        Object {
          "color": "#231F20",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "fontSize": 17.2,
          "lineHeight": 20.885714285714286,
          "paddingHorizontal": 19.65714285714286,
          "paddingVertical": 8.6,
        },
      ]
    }
  >
    Whats New
  </Text>
</View>
`;
